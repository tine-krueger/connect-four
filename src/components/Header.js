import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

Header.propTypes = {
    children: PropTypes.string.isRequired
}

export default function Header({children}) {
    return (
        <HeaderStyled>
            <h1>{children}</h1>
        </HeaderStyled>
    )
}

const HeaderStyled = styled.div`
    text-align: center;
`
