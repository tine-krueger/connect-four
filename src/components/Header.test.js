import React from 'react'
import { render } from '@testing-library/react'
import Header from './Header'
import { expect, it } from '@jest/globals'

describe('Header', () => {
  it('renders correctly', () => {
    const { container } = render(<Header>Your Favourites</Header>)
    expect(container.firstChild).toMatchSnapshot()
  })

  it('shows the correct headline', () => {
      const { getByText } = render(<Header>Your Favourites</Header>)
      expect(getByText(/Your Favourites/i)).toBeInTheDocument()
  })

})