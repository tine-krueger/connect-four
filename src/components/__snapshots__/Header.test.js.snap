// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  class="sc-bdfBwQ eAnEwv"
>
  <h1>
    Your Favourites
  </h1>
</div>
`;
